pkg/api/rest/resttest/resttest.go

```diff
@@ -667,10 +667,6 @@ func (t *Tester) testDeleteGracefulImmediate(obj runtime.Object, setFn SetFunc,
 	if !errors.IsNotFound(err) {
 		t.Errorf("unexpected error, object should be deleted immediately: %v", err)
 	}
-	objectMeta = t.getObjectMetaOrFail(out)
```

Comment:
> Also, this may be a breaking change to clients that were relying on DeletionGracePeriodSeconds to be set - please check the Kubelet signals non-graceful deletion (should give it 2 seconds to delete, which is the minimum) instead of defaulting to the pod spec value.

I don't think this work as expected today. When forcefully deleting a pod, kubelet receives two updates consecutively in a short time frame; the first one is the `DeletionTimestamp` and `DeletionGracePeriodSeconds` update, and the second one is the actual removal of the pod object. Due to kubelet's internal implementation, it's possible that the first update hasn't been processed by the time the removal comes. In this case, the periodic cleanup routine will pick up the work to delete the containers in this pod. At this point, there is no pod spec left and the cleanup routine can only recovers the `TerminationGracePeriodSeconds` from the container label and use that.

We can change the cleanup routine to always use the minimum 2s, assuming all graceful termination will be handled correctly by kubelet itself. We need to re-examine the potential race conditions though.